packages/angular_devkit/schematics/src/tree/virtual.ts

```diff
@@ -49,15 +49,20 @@ export class VirtualDirEntry implements DirEntry {
     return this._path == '/' ? null : this._tree.getDir(dirname(this._path));
   }
   get path() { return this._path; }
+
+  private _getDirectChildren(paths: Path[]) {
+    return paths
+      .filter(path => dirname(path) === this._path)
+      .map(path => basename(path));
```

Comment:
This isn't proper logic. Your DRY has flaws in it.

If `files` is the following:

```js
[
  '/sub1/sub2/sub3/file1',
  '/sub1/sub4/sub5/file2',
  '/sub1/file3',
]
```

Listing `/sub1` should result in `[sub2, sub4, file3]`, but it won't using that new logic (it will list `[file1, file2, file3]`). There is a reason it wasn't DRY; the logic is different when listing files and dirs.

Reply:
Indeed, after diving into the code for the first and investigating hard, I didn't have sufficient resources anymore to design it properly I guess... I'm working on the fix right now :) (and of course on enhancing the current test cases accordingly)